LgiHdmiCec/LgiHdmiCec.cpp

```diff
@@ -0,0 +1,1190 @@
+
+        void LgiHdmiCec::readAddresses()
+        {
+            LOGINFO();
+
+            try
+            {
+              device::VideoOutputPort vPort = device::Host::getInstance().getVideoOutputPort("HDMI0");
+              if (vPort.isDisplayConnected())
+              {
+                  if (getPhysicalAddress() || getLogicalAddress())
+                  {
+                      cecAddressesChanged(PHYSICAL_ADDR_CHANGED);
+                  }
+              }
+            }
+            catch (const std::exception& e)
+            {
+                LOGWARN("exception caught from %s\r\n", __FUNCTION__);
+                throw;
+            }
+        }
+
+        bool LgiHdmiCec::getPhysicalAddress()
+        {
+            bool changed = false;
+            LOGINFO("Entered getPhysicalAddress ");
+
+            uint32_t physAddress = 0x0F0F0F0F;
+
+            try {
+                    LibCCEC::getInstance().getPhysicalAddress(&physAddress);
+
+                    LOGINFO("getPhysicalAddress: physicalAddress: %x %x %x %x ", (physAddress >> 24) & 0xFF, (physAddress >> 16) & 0xFF, (physAddress >> 8)  & 0xFF, (physAddress) & 0xFF);
+                    if (physAddress != physicalAddress)
+                    {
+                        physicalAddress = physAddress;
+                        changed = true;
+                    }
+            }
+            catch (const std::exception& e)
+            {
+                LOGWARN("DS exception caught from getPhysicalAddress");
+                throw;
+            }
+            return changed;
+        }
+
+        bool LgiHdmiCec::getLogicalAddress()
+        {
+            LOGINFO("Entered getLogicalAddress ");
+            bool changed = false;
+
+            try{
+                int addr = LibCCEC::getInstance().getLogicalAddress(DEV_TYPE_TUNER);
+
+                std::string logicalAddrDeviceType = DeviceType(LogicalAddress(addr).getType()).toString().c_str();
+
+                LOGWARN("logical address obtained is %d , saved logical address is %d ", addr, logicalAddress);
+
+                if ((int)logicalAddress != addr || logicalAddressDeviceType != logicalAddrDeviceType)
+
+                {
+                    logicalAddress = addr;
+                    logicalAddressDeviceType = logicalAddrDeviceType;
+                    changed = true;
+                }
+            }
+            catch (const std::exception& e)
+            {
+                LOGWARN("CEC exception caught from getLogicalAddress ");
+                throw;
+            }
+
+            return changed;
+        }
+
+        bool LgiHdmiCec::getEnabled()
+        {
+            LOGWARN("Entered getEnabled ");
+            if(true == cecEnableStatus)
+                return true;
+            else
+                return false;
+        }
+
+        std::string LgiHdmiCec::getName()
+        {
+            //SVCLOG_WARN("%s \r\n",__FUNCTION__);
+            IARM_Result_t ret = IARM_RESULT_INVALID_STATE;
```

Comment:
don't understand flow of this function, can we simplify that?

Reply:
This is what the original HdmiCec.cpp file looks like here. Should I clean this up in the original and upstream the change?